flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/benchmark/scheduling/StartSchedulingInBatchJobBenchmarkTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+package org.apache.flink.runtime.scheduler.benchmark.scheduling;
+
+import org.apache.flink.runtime.scheduler.benchmark.JobConfiguration;
+import org.apache.flink.runtime.scheduler.strategy.SchedulingStrategy;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+/**
+ * The benchmark of starting scheduling in a BATCH job. The related method is {@link
+ * SchedulingStrategy#startScheduling}.
+ */
+public class StartSchedulingInBatchJobBenchmarkTest extends TestLogger {
+
+    @Test
+    public void startSchedulingInBatchJobBenchmark() throws Exception {
+        StartSchedulingInBatchJobBenchmark benchmark = new StartSchedulingInBatchJobBenchmark();
+        benchmark.setup(JobConfiguration.BATCH_TEST);
+        benchmark.startScheduling();
+        benchmark.teardown();
+    }
+
+    @Test
+    public void startSchedulingInAdaptiveBatchJobBenchmark() throws Exception {
+        StartSchedulingInBatchJobBenchmark benchmark = new StartSchedulingInBatchJobBenchmark();
+        benchmark.setup(JobConfiguration.ADAPTIVE_BATCH_TEST);
+        benchmark.startScheduling();
+        benchmark.teardown();
+    }
+
+    @Test
+    public void startSchedulingInAdaptiveBatchHybridJobBenchmark() throws Exception {
```

Comment:
Maybe we also should add a test for hybrid corresponds to a non-dynamic graph, which means that the downstream has been initialized during upstream scheduling.

Reply:
Thanks for your suggestions! Do you mean a batch job in AdaptiveBatch mode whose parallelism is user-specified? If so, current test has satisfied the case 'the downstream has been initialized during upstream scheduling'.